compiler/rustc_attr_parsing/src/session_diagnostics.rs

```diff
@@ -610,20 +610,33 @@ pub(crate) enum AttributeParseErrorReason<'a> {
     ExpectedIdentifier,
 }
 
+/// A description of a thing that can be parsed using an attribute parser.
+#[derive(Copy, Clone)]
+pub enum ParsedDescription {
+    Attribute,
+    Macro,
```

Comment:
could you comment on the reason why this is needed? It's not obvious that we're using an attribute parser for non-attributes. r=me after that